tests/pkg_yxml/main.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2020 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ * @file
+ * @brief       Tests for the yxml XML parser library package
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @}
+ */
+#include <string.h>
+#include <stdio.h>
+
+#include "embUnit.h"
+#include "yxml.h"
+
+static char _yxml_buf[1024];
+
+static const char *_doc = "<foo>bar</foo>";
+
+static void test_yxml(void)
+{
+    yxml_t yxml;
+    char content[64];
+
+    /* initialize yxml instance */
+    yxml_init(&yxml, _yxml_buf, sizeof(_yxml_buf));
+
+    /* parse XML document bytewise */
+    for (const char *doc = _doc; *doc; doc++) {
+        yxml_ret_t r = yxml_parse(&yxml, *doc);
+        TEST_ASSERT(r >= 0);
+
+        /* handle tokens */
+        switch (r) {
+            case YXML_OK:
+                break;
+            case YXML_ELEMSTART:
+                printf("YXML_ELEMSTART \"%s\"\n", yxml.elem);
+                TEST_ASSERT_EQUAL_INT(0, strcmp("foo", yxml.elem));
```

Comment:
```suggestion
                TEST_ASSERT_EQUAL_STRING("foo", yxml.elem);
```
btw ;-) then you also get the output in case `yxml.elem` is wrong ;-)